symfony/plugins/orangehrmTimePlugin/Dao/ProjectDao.php

```diff
@@ -212,4 +215,90 @@ public function hasTimesheetItemsForProject(int $projectId): bool
         $qb->setParameter('projectId', $projectId);
         return $this->getPaginator($qb)->count() > 0;
     }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @return array
+     */
+    public function getProjectReportCriteriaList(ProjectReportSearchFilterParams $projectReportSearchFilterParams
+    ): array {
+        return $this->getProjectReportCriteria($projectReportSearchFilterParams)->getQuery()->execute();
+    }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @return int
+     */
+    public function getProjectReportCriteriaListCount(ProjectReportSearchFilterParams $projectReportSearchFilterParams
+    ): int {
+        return $this->getProjectReportCriteria($projectReportSearchFilterParams)->count();
+    }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @return Paginator
+     */
+    private function getProjectReportCriteria(ProjectReportSearchFilterParams $projectReportSearchFilterParams
+    ): Paginator {
+        $q = $this->createQueryBuilder(TimesheetItem::class, 'timesheetItem');
+        $q->select(
+            'projectActivity.id AS activityId, projectActivity.name, projectActivity.deleted AS deleted, SUM(COALESCE(timesheetItem.duration, 0)) AS totalDuration'
+        );
+        $q->leftJoin('timesheetItem.projectActivity', 'projectActivity');
+        $q->leftJoin('timesheetItem.timesheet', 'timesheet');
+        $this->setSortingAndPaginationParams($q, $projectReportSearchFilterParams);
+
+        $this->getCommonQuery($projectReportSearchFilterParams, $q);
+        $q->groupBy('projectActivity.id');
+
+        return $this->getPaginator($q);
+    }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @return string
+     */
+    public function getTotalDurationForProjectReport(ProjectReportSearchFilterParams $projectReportSearchFilterParams): string
+    {
+        $q = $this->createQueryBuilder(TimesheetItem::class, 'timesheetItem');
+        $q->select('SUM(COALESCE(timesheetItem.duration, 0)) AS totalDuration');
+        $q->leftJoin('timesheetItem.projectActivity', 'projectActivity');
+        $q->leftJoin('timesheetItem.timesheet', 'timesheet');
+
+        $this->getCommonQuery($projectReportSearchFilterParams, $q);
```

Comment:
You can try this, to remove `use Doctrine\ORM\QueryBuilder;`
```suggestion
        $this->getCommonQuery($projectReportSearchFilterParams, $this->getQueryBuilderWrapper($q));
```
Ref: https://github.com/orangehrm/orangehrm/blob/5.0/symfony/plugins/orangehrmPimPlugin/Dao/EmployeeDao.php#L163

Reply:
done